mantis-control-plane/mantis-control-plane-client/src/main/java/io/mantisrx/server/master/client/HighAvailabilityServicesUtil.java

```diff
@@ -67,5 +79,48 @@ public MantisMasterGateway getMasterClientApi() {
     public MasterMonitor getMasterMonitor() {
       return curatorService.getMasterMonitor();
     }
+
+    @Override
+    public ResourceLeaderConnection<ResourceClusterGateway> connectWithResourceManager(
+        ClusterID clusterID) {
+      return new ResourceLeaderConnection<ResourceClusterGateway>() {
+        final MasterMonitor masterMonitor = curatorService.getMasterMonitor();
+
+        ResourceClusterGateway currentResourceClusterGateway =
+            new ResourceClusterGatewayClient(clusterID, masterMonitor.getLatestMaster());
+
+        final String nameFormat =
+            "ResourceClusterGatewayCxn (" + rmConnections.getAndIncrement() + ")-%d";
+        final Scheduler scheduler =
+            Schedulers
+                .from(
+                    Executors
+                        .newSingleThreadExecutor(
+                            new ThreadFactoryBuilder().setNameFormat(nameFormat).build()));
+
+        final List<Subscription> subscriptions = new ArrayList<>();
```

Comment:
Where do we use these subscriptions?

Reply:
Nowhere right now. But, I think the intention when I wrote the code (a long time ago) was that we would close these subscriptions when the task executor is about to be shut down. So, it just makes for a clean, graceful shutdown.